app/scripts/ViewConfigEditor.js

```diff
@@ -94,7 +128,35 @@ class ViewConfigEditor extends React.Component {
     this.setState({ hide: false });
   }
 
+  hideLog() {
+    this.setState({ showLog: false });
+  }
+
+  showLog() {
+    this.setState({ showLog: true });
+  }
+
+  toggleLog() {
+    if (this.state.showLog) {
+      this.hideLog();
+    } else {
+      this.showLog();
+    }
+  }
+
   render() {
+    const logMessages = this.state.logMsgs.map((d, i) => {
+      const key = `${i}-${d.type}`;
```

Comment:
These keys may not be unique.

For example, you do something and get  one  error message `1-Warning`. Then you fix that error  and get a different warning which would now also be `1-Warning` but may have a different `msg`.

Reply:
You are right, and thanks for the explanation! I've replaced it to `d.msg`.